input-profile.js

```diff
@@ -0,0 +1,164 @@
+					visualResponses = components[i]["visualResponses"];
+					for (j in visualResponses) {
+						if (visualResponses.hasOwnProperty(j)) {
+							this.getObjectName(obj, visualResponses[j]["valueNodeName"]);
+							this.getObjectName(obj, visualResponses[j]["minNodeName"]);
+							this.getObjectName(obj, visualResponses[j]["maxNodeName"]);
+						}
+					}
+				}
+			}
+		},
+
+		getObjectName: function (obj, name) {
+			if (obj.name == name) this.gamepadObjects[name] = obj;
+			const children = obj.children;
+			for (let i = 0; i < children.length; ++i) this.getObjectName(children[i], name);
+		},
+
+		assignTransform: function (target, min, max, value) {
+			const tempVec = glMatrix.vec3.create();
+			glMatrix.vec3.lerp(
+				tempVec,
+				min.getTranslationWorld([]),
+				max.getTranslationWorld([]),
+				value,
+			);
+			target.setTranslationWorld(tempVec);
+
+			const tempquat = glMatrix.quat.create();
+			glMatrix.quat.lerp(tempquat, min.rotationWorld, max.rotationWorld, value);
+			glMatrix.quat.normalize(tempquat, tempquat);
+			target.rotationWorld = tempquat;
+		},
+
+		mapGamepadInput: function () {
+			components = ProfileJSON["layouts"][this.handedness]["components"];
+			for (i in components) {
+				if (components.hasOwnProperty(i)) {
+					component = components[i];
+					visualResponses = component["visualResponses"];
+					for (j in visualResponses) {
+						if (visualResponses.hasOwnProperty) {
+							visualResponse = visualResponses[j];
+							let target = this.gamepadObjects[visualResponse["valueNodeName"]];
+							let min = this.gamepadObjects[visualResponse["minNodeName"]];
+							let max = this.gamepadObjects[visualResponse["maxNodeName"]];
+							this.assignTransform(
+								target,
+								min,
+								max,
+								this.getGamepadValue(component, visualResponse),
+							);
+						}
+					}
+				}
+			}
+		},
+
+		getGamepadValue: function (component, visualResponse) {
+			if (visualResponse["valueNodeProperty"] == "transform") {
+				switch (component["type"]) {
+					case "button":
+						var value = this.gamepad.buttons[component["gamepadIndices"]["button"]].pressed;
+						break;
+					case "thumbstick":
+						// check if component property matches with axis
+						if (visualResponse["componentProperty"] == "button") {
+							var value = this.gamepad.buttons[component["gamepadIndices"]["button"]].pressed;
+						}
+						if (visualResponse["componentProperty"] == "xAxis") {
+							var value = (this.gamepad.axes[component["gamepadIndices"]["xAxis"]] + 1) / 2;
+						}
+						if (visualResponse["componentProperty"] == "yAxis") {
+							var value = (this.gamepad.axes[component["gamepadIndices"]["yAxis"]] + 1) / 2;
+						}
+						break;
+					case "trigger":
+						var value = this.gamepad.buttons[component["gamepadIndices"]["button"]].value;
+						break;
+
+					case "squeeze":
+						var value = this.gamepad.buttons[component["gamepadIndices"]["button"]].value;
+						break;
+				}
+			}
```

Comment:
I suggest you avoid the `break;` and `var value` and just return the value directly with `return this.gamepad.buttons[...`, will result in much simpler code.

And the final return should probably be a useful value, either `null` or `0`, depending on how the calling code is handling those cases where the visualResponse is something else or the component type is unknown.